crates/amaru/src/bin/amaru/cmd/fetch_chain_headers.rs

```diff
@@ -0,0 +1,265 @@
+// Copyright 2025 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::cmd::{DEFAULT_NETWORK, connect_to_peer};
+use amaru::stages::pull;
+use amaru_consensus::IsHeader;
+use amaru_kernel::{Header, Point, from_cbor, network::NetworkName, peer::Peer};
+use amaru_network::session::PeerSession;
+use amaru_progress_bar::{ProgressBar, new_terminal_progress_bar};
+use clap::{Parser, arg};
+use gasket::framework::{AsWorkError, WorkerError};
```

Comment:
We don't want gasket dependency here

Reply:
I must admit, I just copy/pasted what we had in import_headers without giving it a lot of though. I'm not sure why but it has this dependency. Maybe I can see how to replace the code to not use gasket here?